### ISSUE TITLE
Threema Version 4 - DB-Update

### ISSUE THREAD

**Issue by moffer** [greenify/threema-decrypt#2]
Since Threema Version 4 it is using also an updated version of sqlcipher (version 4). Therefore also some other configurations of sqlcipher has been updated. 
Instead of `PRAGMA cipher_default_kdf_iter = 4000;...` following code is needed:

```
PRAGMA cipher_default_kdf_iter = 1;
PRAGMA key='$key';
PRAGMA kdf_iter = 1;
PRAGMA cipher_memory_security = OFF;
```
Otherwise it says `file is not a database`.

**Comment by mrthaggar** [comment]
Have you been able to read the contents of the threema4.db using this tool?

**Comment by ohunecker** [comment]
Yes, you can decrypt content from threema4.db with this tool. All you need is sqlclipher version 4.
Older versions only show the error:

> error: file is encrypted or is not a database

You can check your local version with
`PRAGMA cipher_version;`

Build a newer version from sqlclipher from [source](https://github.com/sqlcipher/sqlcipher/) if needed and use the PRAGMA-Params from moffer's post to decrypt database.
